pkg/api/api.go

```diff
@@ -5,26 +5,35 @@ import (
 	"net/http"
 )
 
-// result contains the information generated by and endpoint handler.
-type result struct {
-	Output     interface{} `json:"output,omitempty"`
-	ErrCode    int         `json:"err_code"`
-	ErrMessage string      `json:"err_message,omitempty"`
+// Result contains the information generated by and endpoint handler.
+type Result struct {
+	Errors []Error     `json:"errors,omitempty"`
+	Data   interface{} `json:"data,omitempty"`
+}
+
+type Error struct {
+	Code    int    `json:"err_code"`
+	Message string `json:"err_message,omitempty"`
```

Comment:
Nitpick: apparently all these identifiers are not exported from `api` package, so they should  start with a lowercase letter. In any case, please write a documentation for `Error` so the tests pass.